CONFIGURATION.md

```diff
@@ -0,0 +1,279 @@
+
+
+### Proposal: Temporary Strains
+
+We introduce a new option for `hlx deploy --temp <tag[]?>` that takes zero or more tags for temporary deployments. By default, it uses `CIRCLE_SHA1` as the tag, i.e. the SHA1 hash of the last commit of the current build. This ties the deployment to the individual commit and avoids accidential overwriting.
+
+When running `hlx deploy --temp`, all strains that refer a code repository that differs from the current repository will be ignored to avoid unccessary deploys.
+All strains that refer to the current code repository will be deployed with a `code` prefix that ends with the value of `tag`, e.g. `/acapt/default/https---github-com-adobe-project-helix-io-git--master--` would become `/acapt/default/https---github-com-adobe-project-helix-io-git--35ef52a772a7656be3d31527f8e595e4e286a0d4--`. 
+Each strain that is thus modified will be transformed into a temporary strain where the name includes the old strain name and the new `tag`, e.g. `default/35ef52a772a7656be3d31527f8e595e4e286a0d4`, `client/35ef52a772a7656be3d31527f8e595e4e286a0d4`, `pipeline/35ef52a772a7656be3d31527f8e595e4e286a0d4`. 
+These modified strains will not be written into `.hlx/strains.json` or persisted in any other way.
+Instead, the strain resolution logic in VCL will be modified so that when an `X-Strain` cookie or header is present and the value of the header contains a `/`, both the `X-Strain` (name) and `X-Tag` (tag) will be parsed. 
+For the most part, the default logic of the current `X-Strain` will be applied, so that the strain's directory index, static repo, etc. will be used. The only exception is the resolution of the OpenWhisk action to execute.
+Here, the `-git--([\w]+)--` pattern will be replaced with the `X-Tag` value, effectively pinning the used action to the tagged deployment. 
+
+As this will lead to a large number of temporary actions in OpenWhisk, we also introduce a `hlx undeploy` command that clears a temporary deployment made earlier. This can be run at the end of every CI job.
+
+`hlx test` and `hlx perf` should use the new temporary strains when running in CI.
+
+## Discussion: layout
+
+During the 4/2018 hackathon we also discussed the layout of the project. it was decided that:
+
+- templates and pre functions will be in `./src`
+- client side javascript **must not** go into `./src`. 
+- we will provide a react example, and put the sources in `./react`
+- all static files will go to `htdocs`
+- all additional openwhisk actions will go to `htdocs/cgi-bin`
+
+## Discussion: the `code` property
+
+- the `code` property of a strain is either a giturl string or giturl object (but not an action name).
```

Comment:
I'm still not sure how this works.

so assume a non-dirty checkout, `git@github.com/adobe/helix-io.git#master (923c1e6)`.
1. the default strain contains a code property: `code: "git@github.com/adobe/helix-io.git#master"`
2. `hlx deploy` would generate an action `/helix/923c1e6/html`
3. and `effective-helix-config.yaml` would then have a property: `code: "git@github.com/adobe/helix-io.git#923c1e6"`

assume a dirty checkout, `git@github.com/adobe/helix-io.git#master (a0c96a5)`
1. the default strain contains a code property: `code: "git@github.com/adobe/helix-io.git#master"`
2. `hlx deploy --dirty` would generate an action `/helix/a0c96a5-dirty/html`
3. and `effective-helix-config.yaml` would then have a property: `code: "git@github.com/adobe/helix-io.git#a0c96a50-dirty"`<sup>1</sup>

<sup>1</sup> not sure about the `-dirty` suffix in the ref here. but fastly wouldn't have a way to figure it our